src/main/java/org/jitsi/jicofo/ParticipantChannelAllocator.java

```diff
@@ -217,10 +217,12 @@ else if (!canceled)
         if (expireChannels || canceled)
         {
             // Whether another thread intentionally canceled us, or there was
-            // a failure to invite the participant in the jingle level, we will
-            // not trigger a retry here.
-            // In any case, try and expire the channels on the bridge.
-            bridgeSession.terminate(participant);
+            // a failure to invite the participant on the jingle level, we will
+            // not trigger a retry here. To assure that the conference doesn't
+            // attempt a retry, we pretend that this channel allocator is for
+            // a reinvite.
+            reInvite = true;
```

Comment:
It used to be that the reInvite flag's meaning is to distinguish between the Jingle message that needs to go to the participant ('session-initiate' vs 'transport-replace'). Wouldn't it be less confusing to come up with a separate flag which meaning would be clear that it's about "do not retry to allocate channels" instead of mixing up the two concepts together ?

Reply:
Sure ;)